src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -1186,6 +1187,11 @@ public boolean isFlag(){
             return field.getType().equals(boolean.class) || field.getType().equals(Boolean.class);
         }
 
+        /** Returns {@code true} if the argument has a bounded range; {@code false} otherwise. */
+        public boolean hasBoundedRange() {
+            return this.minValue != Double.NEGATIVE_INFINITY || this.maxValue != Double.POSITIVE_INFINITY;
+        }
+
```

Comment:
I think this could be private for now. Also, now that this method exists, we should use it in the other places where we do the same check (i.e., in the argdef constructor). Likewise, we should add a hasRecommendedRange and use that at the appropriate call sites.